cli_tools/gce_vm_image_import/importer/bootable_disk_processor.go

```diff
@@ -16,37 +16,103 @@ package importer
 	daisy_utils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
 	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/compute/v1"
 )
 
 type bootableDiskProcessor struct {
-	workflow        *daisy.Workflow
-	userLabels      map[string]string
-	storageLocation string
-	uefiCompatible  bool
-	noExternalIP    bool
-	network         string
-	OS              string
+	ImportArguments
+	computeClient daisyCompute.Client
+	diskInspector disk.Inspector
+	workflow      *daisy.Workflow
+	pd            persistentDisk
 }
 
-func (b bootableDiskProcessor) process() (err error) {
+func (b bootableDiskProcessor) process() (persistentDisk, error) {
+	var err error
+	pd, err := b.inspectAndPreProcess()
+	if err != nil {
+		return pd, err
+	}
+
 	err = b.workflow.RunWithModifiers(context.Background(), b.preValidateFunc(), b.postValidateFunc())
 	if err != nil {
-		daisy_utils.PostProcessDErrorForNetworkFlag("image import", err, b.network, b.workflow)
+		daisy_utils.PostProcessDErrorForNetworkFlag("image import", err, b.Network, b.workflow)
 		err = customizeErrorToDetectionResults(b.OS,
 			b.workflow.GetSerialConsoleOutputValue("detected_distro"),
 			b.workflow.GetSerialConsoleOutputValue("detected_major_version"),
 			b.workflow.GetSerialConsoleOutputValue("detected_minor_version"), err)
 	}
-	return err
+	return pd, err
+}
+
+func (b bootableDiskProcessor) inspectAndPreProcess() (persistentDisk, error) {
+	if !b.Inspect || b.diskInspector == nil {
+		return b.pd, nil
+	}
+
+	ir, err := b.inspectDisk()
+	if err != nil {
+		return b.pd, err
+	}
+
+	return b.recreateDisk(ir)
+}
+
+func (b bootableDiskProcessor) recreateDisk(ir disk.InspectionResult) (persistentDisk, error) {
+	// If UEFI_COMPATIBLE is enforced in user input args (by d.ImportArguments.UefiCompatible),
+	// then it has been honored in inflation stage, so no need to create a new disk here.
+	// Create new disk with UEFI_COMPATIBLE only when inspection result tells us to do.
+	if !b.UefiCompatible && ir.HasEFIPartition {
+		diskName := fmt.Sprintf("disk-%v-uefi", b.ExecutionID)
+		err := b.computeClient.CreateDisk(b.Project, b.Zone, &compute.Disk{
+			Name:            diskName,
+			SourceDisk:      b.pd.uri,
+			GuestOsFeatures: []*compute.GuestOsFeature{{Type: "UEFI_COMPATIBLE"}},
+		})
+		if err != nil {
+			return b.pd, daisy.Errf("Failed to create UEFI disk: %v", err)
+		}
+		log.Println("UEFI disk created: ", diskName)
+
+		// Cleanup the old disk after the new disk is created.
+		cleanupDisk(b.computeClient, b.Project, b.Zone, b.pd)
+
+		b.pd.uri = fmt.Sprintf("zones/%v/disks/%v", b.Zone, diskName)
+	}
+
+	if b.UefiCompatible || ir.HasEFIPartition {
+		b.pd.isUEFICompatible = true
+	}
+
+	if ir.HasEFIPartition {
+		b.pd.isUEFIDetected = true
+	}
```

Comment:
Here's your updating copies of b.pd because both b and pd are passed as values, not references. The tests are failing for other reasons, but I assume it was not your intention to modify copies of objects since these changes will be lost once func ends.